app/core/Engine/messengers/multichain-accounts-service-messenger/multichain-accounts-service-messenger.ts

```diff
@@ -0,0 +1,70 @@
+import { Messenger } from '@metamask/base-controller';
+import {
+  AccountsControllerAccountAddedEvent,
+  AccountsControllerAccountRemovedEvent,
+  AccountsControllerGetAccountAction,
+  AccountsControllerGetAccountByAddressAction,
+  AccountsControllerListMultichainAccountsAction,
+} from '@metamask/accounts-controller';
+import { HandleSnapRequest as SnapControllerHandleRequest } from '@metamask/snaps-controllers';
+import {
+  KeyringControllerWithKeyringAction,
+  KeyringControllerGetStateAction,
+  KeyringControllerStateChangeEvent,
+} from '@metamask/keyring-controller';
+
+type Actions =
+  | AccountsControllerListMultichainAccountsAction
+  | AccountsControllerGetAccountAction
+  | AccountsControllerGetAccountByAddressAction
+  | SnapControllerHandleRequest
+  | KeyringControllerGetStateAction
+  | KeyringControllerWithKeyringAction;
+
+type Events =
+  | KeyringControllerStateChangeEvent
+  | AccountsControllerAccountAddedEvent
+  | AccountsControllerAccountRemovedEvent;
+
+export type MultichainAccountServiceMessenger = ReturnType<
+  typeof getMultichainAccountServiceMessenger
+>;
+
+/**
+ * Get a restricted messenger for the account wallet controller. This is scoped to the
```

Comment:
I might have missed that one on the extension too 😅

```suggestion
 * Get a restricted messenger for the multichain account service. This is scoped to the
```